### PR TITLE
updated to online Json

### PR DESCRIPTION
**Pull request by lakshminaidu**
Updated the code to online Json

### CODE REVIEW

**Comment by simonzhexu** [comment]
@lakshminaidu thanks for sending pull request, please complete CLA as mentioned in the CONTRIBUTIONS.md

File: iOS/WAStickersThirdParty/JSONDownLoader.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//  Created by Poornima Sreedhar on 26/11/18.
+//  Copyright © 2018 WhatsApp. All rights reserved.
+//
+
+import Foundation
+enum Result<Value> {
+    case success(Value)
+    case failure(Error)
+}
+class JSONDownLoader {
```
Comment (Zandor300) [inline iOS/WAStickersThirdParty/JSONDownLoader.swift]: I personally think it should be JSONDownloader.

File: iOS/WAStickersThirdParty/AllStickerPacksViewController.swift [hunk 2]
```diff
@@ -39,78 +39,125 @@ class AllStickerPacksViewController: UIViewController, UITableViewDataSource, UI
...
+        catch {
+            fatalError("sticker_packs.wasticker not found.")
+        }
+        
+    }
+
+
 
-    private func fetchStickerPacks() {
+//    private func fetchStickerPacks() {
```
Comment (Zandor300) [inline iOS/WAStickersThirdParty/AllStickerPacksViewController.swift]: I think if this gets merged into the master branch, we should at least have both and allow developers to choose which route they want to provide their sticker packs.

File: iOS/WAStickersThirdParty/AllStickerPacksViewController.swift [hunk 3]
```diff
@@ -8,12 +8,12 @@
...
 import UIKit
 
-class AllStickerPacksViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+class AllStickerPacksViewController: UIViewController {
 
     @IBOutlet private weak var stickerPacksTableView: UITableView!
 
     private var needsFetchStickerPacks = true
-    private var stickerPacks: [StickerPack] = []
+    private var stickerPacks: [WAStickerPack] = []
```
Comment (Zandor300) [inline iOS/WAStickersThirdParty/AllStickerPacksViewController.swift]: Weird how you changed all `StickerPack` references to `WAStickerPack` and `Sticker` to `WASticker` but when looking at your fork, it is still `StickerPack` and `Sticker`.

Comment (Zandor300) [reply iOS/WAStickersThirdParty/AllStickerPacksViewController.swift]: Also, is there a reason behind the renaming? I personally think `Sticker` and `StickerPack` are just fine. It is only a WhatsApp sticker app after all.

File: iOS/WAStickersThirdParty/StickerCell.swift [hunk 4]
```diff
@@ -12,26 +12,33 @@ final class StickerCell: UICollectionViewCell {
...
-                        }
-                    }
-                }
+            if let imageFile = sticker?.imageFile {
+                self.imageView.image = UIImage(named: imageFile)
             }
+
         }
+
+//        willSet {
```
Comment (Zandor300) [inline iOS/WAStickersThirdParty/StickerCell.swift]: Again, if this is going to be merged, it should at least support both. Maybe add an if-statement checking if it is a sticker from the web or from the app bundle.

File: iOS/WAStickersThirdParty/WAStickerPackResponse.swift [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        case publisher
+        case trayImage = "tray_image_file"
+        case publisherWebsite = "publisher_website"
+        case privacyPolicyWebsite = "privacy_policy_website"
+        case licenseAgreementWebsite = "license_agreement_website"
+        case stickers
+        
+    }
+    
+    var imageData: ImageData?  {
```
Comment (Zandor300) [inline iOS/WAStickersThirdParty/WAStickerPackResponse.swift]: Maybe rename this to `trayImageData`. Currently it is a bit vague that this is the `ImageData` of the `trayIcon`.

**Comment by albrye** [comment]
Add an additional review comment

**Comment by simonzhexu** [comment]
Online JSON is not a way we typically handle this. So I would not be able to pick this change into the master branch.

**Pull request closed without merging**